app/controllers/wikis_controller.rb

```diff
@@ -0,0 +1,65 @@
+class WikisController < ApplicationController
+  before_action :authenticate_user!, except: [:show, :index]
+
+  def new
+    @wiki = Wiki.new
+  end
+
+  def index
+    @wikis = Wiki.all.order(created_at: :desc)
+  end
+
+  def create
+    @wiki = current_user.wiki.build(wiki_params)
+    @wiki.user = current_user
+
+    if @wiki.save
+      redirect_to @wiki
+    else
+      flash.now[:alert] = "Unable to save wiki. Please try again later"
+      render :new
+    end
+  end
+
+  def show
+    begin
+      @wiki = Wiki.find(params[:id])
+    rescue ActiveRecord::RecordNotFound => e
+      redirect_to wikis, flash.now[:error] = "Whoops! Could not find that wiki. \n #{e.message}"
+    end
+  end
+
+  def update
+    @wiki = Wiki.find(params[:id])
+    @wiki.assign_attributes(wiki_params)
+
+    if @wiki.save
+      flash[:notice] = "Wiki was updated successfully."
+      redirect_to @wiki
+    else
+      flash.now[:alert] = "Unable to save wiki. Please try again later."
+      render :edit
+    end
+  end
+
+  def edit
+    @wiki = Wiki.find(params[:id])
```

Comment:
Any reason why you rescue RecordNotFound in the `show` action but not in this one?